### DISCUSSION TITLE
Error Executing Timertask

### DISCUSSION THREAD

**Discussion by patrick-tix** [camueller/SmartApplianceEnabler#363] [Allgemein]
Hallo Zusammen,

ich habe mit einer meiner Tasmota Steckdosen (für einen Trockner) ein Problem beim Auslesen der aktuellen Leistung.
Der Verbrauch wurde gestern während des Trockner-Laufs ordentlich aufgezeichnet und an das Sunny Portal übermittelt.
Jedoch wurde seit Ende des Laufs immer der gleiche Verbrauch (ca. 50W) übermittelst.

Im Log habe ich nun minütlich folgenden Fehler gefunden:

`2022-10-12 08:34:16,748 DEBUG [Timer-0] d.a.s.h.HttpHandler [HttpHandler.java:99] F-20181211-000000000003-00: url=http://192.168.2.63/cm?cmnd=Status 8 method=GET data=null path=null

2022-10-12 08:34:16,748 DEBUG [Timer-0] d.a.s.h.HttpHandler [HttpHandler.java:101] F-20181211-000000000003-00: Response: {"StatusSNS":{"Time":"2022-10-12T07:34:15","ENERGY":{"TotalStartTime":"2021-11-22T14:14:17","Total":3.161,"Yesterday":0.886,"Today":0.000,"Power": 0,"ApparentPower": 0,"ReactivePower": 0,"Factor":0.00,"Voltage":223,"Current":0.000}}}

2022-10-12 08:34:16,748 ERROR [Timer-0] d.a.s.u.GuardedTimerTask [GuardedTimerTask.java:60] F-20181211-000000000003-00: Error executing timer task name=PollPowerMeter id=8441107

2022-10-12 08:34:17,103 WARN [Timer-0] d.a.s.c.MeterReportingSwitch [MeterReportingSwitch.java:170] F-20181211-000000000003-00: Meter not set.`

Die Einstellung ist die gleiche wie für die anderen Tasmota Steckdosen, die einwandfrei funktionieren.
![tasmota_setting](https://user-images.githubusercontent.com/115613948/195269152-d8c55e7a-a3e7-466f-bdc7-dd182629406d.png)

Habt ihr eine Idee, woran es liegen könnte und was ich noch prüfen kann?

**Comment by camueller** [comment]
Welche SAE-Version verwendest Du? Scheint ein 2.x zu sein ...

Du schreibst, dass Du mehrere Tasmota-Geräte hast, die fehlerfrei funktionieren und eines nicht? Unterscheidet sich die Konfiguration der Zähler?

Die Meldung:
```
2022-10-12 08:34:17,103 WARN [Timer-0] d.a.s.c.MeterReportingSwitch [MeterReportingSwitch.java:170] F-20181211-000000000003-00: Meter not set.
```
deutet darauf hin, das kein Zähler konfiguriert wurde.

**Comment by patrick-tix** [comment]
Ich nutze die 2.0.6

Ich habe  drei Tasmota Geräte mit gleicher Firmware und gleicher Konfiguration (bis auf IP-Adresse) im SAE angelegt.

Der Screenshot oben ist von der Zähler-Konfiguration der betroffenen Trockner-Steckdose.

Während des Trockner-Vorgangs wurde ja auch der Korrekte Wert ausgelesen. Nur danach ist er bei einem Wert stecken geblieben.
Wenn ich die Trockner-URL öffne, wird der derzeitige korrekte Wert angezeigt (im Moment 0).

**Comment by camueller** [comment]
Starte den SAE mal neu. Wenn das Problem dann immer noch auttritt, benötige ich ein Log-File.

Übrigens würde ich bei Tasmota-Geräten bei der Zählekonfiguration eher mit Parameter "Zählerstand" als mit "Leistung" arbeiten.
Ausserdem JSON zum parsen der Response verwenden:

![image](https://user-images.githubusercontent.com/4489628/195300764-890130e2-c2c6-49c2-82b1-e832c02e131f.png)

**Comment by patrick-tix** [comment]
Der Neustart hat nicht geholfen.
Aber ich habe nun nach deiner Empfehlung die Einstellung auf Zählerstand und JSON geändert. Jetzt erscheint kein Fehler mehr im LOG.
Dann werde ich nachher mal den Trockner anwerfen und testen.

`2022-10-12 11:14:05,999 DEBUG [Timer-0] d.a.s.h.HttpHandler [HttpHandler.java:101] F-20181211-000000000003-00: Response: {"StatusSNS":{"Time":"2022-10-12T10:14:05","ENERGY":{"TotalStartTime":"2021-11-22T14:14:17","Total":3.161,"Yesterday":0.886,"Today":0.000,"Power": 0,"ApparentPower": 0,"ReactivePower": 0,"Factor":0.00,"Voltage":224,"Current":0.000}}}`

`2022-10-12 11:14:05,999 DEBUG [Timer-0] d.a.s.h.HttpHandler [HttpHandler.java:67] F-20181211-000000000003-00: value=3.161 protocolHandlerValue=3.161 valueExtractionRegex=null extractedValue=null factorToValue=null`

`2022-10-12 11:14:06,000 DEBUG [Timer-0] d.a.s.m.PollEnergyMeter [PollEnergyMeter.java:105] F-20181211-000000000003-00: Calculating power from energy: power=0 currentEnergyCounter=3.161 previousEnergyCounter=3.161 diffEnergy=0 diffTime=60000`

Vielen Dank schonmal!

**Reply by camueller** [reply]
Wenn Du den Parameter "Zählerstand" verwendest, musst Du die Genauigkeit von Tasmota auf 5 Nachkommastellen erhöhen, sonst wird die Leistung nur ungenau berechnet. Siehe https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/Tasmota_DE.md

**Comment by patrick-tix** [comment]
Danke für die Info, habe ich soeben bei allen Tasmota-Geräten geändert.

Noch zur Info bzgl. des Fehlers: Es gibt doch einen Unterschied zwischen den Tasmota-Geräten.
Der Trockner hat bereits Tasmota 10.0, die anderen beiden Geräte noch Tasmota 6.5. 
Vielleicht wurde da etwas an der Firmware geändert, so dass das Auslesen Probleme macht.
